app/upgrades/v27/upgrades.go

```diff
@@ -24,6 +26,19 @@ func CreateUpgradeHandler(
 			return nil, err
 		}
 
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+		err = InitializeConstitutionCollection(sdkCtx, *keepers.GovKeeper)
+		if err != nil {
+			sdkCtx.Logger().Error("Error initializing Constitution Collection:", "message", err.Error())
+		}
+
 		return migrations, nil
 	}
 }
+
+// setting the default constitution for the chain
+// this is in line with cosmos-sdk v5 gov migration: https://github.com/cosmos/cosmos-sdk/blob/v0.50.10/x/gov/migrations/v5/store.go#L57
+func InitializeConstitutionCollection(ctx sdk.Context, govKeeper govkeeper.Keeper) error {
+	return govKeeper.Constitution.Set(ctx, "This chain has no constitution.")
+}
```

Comment:
Does this imply none of the v5 gov migrations were run? Do we need to do the others as well?

Reply:
The migration on v0.50 didn't function as expected eg

```
    50:	// setting the default constitution for the chain
    51:	// this is in line with cosmos-sdk v5 gov migration: https://github.com/cosmos/cosmos-sdk/blob/v0.50.10/x/gov/migrations/v5/store.go#L57
    52:	func InitializeConstitutionCollection(ctx sdk.Context, govKeeper govkeeper.Keeper) error {
    53:		ok, err := govKeeper.Constitution.Has(ctx)
=>  54:		if !ok || err != nil {
    55:			if err := govKeeper.Constitution.Set(ctx, "This chain has no constitution."); err != nil {
    56:				return err
    57:			}
    58:		}
    59:	
(dlv) p ok
true
(dlv) p err
error nil
(dlv) s
> github.com/osmosis-labs/osmosis/v26/app/upgrades/v27.InitializeConstitutionCollection() ./upgrades.go:60 (PC: 0x49c4319)
    55:			if err := govKeeper.Constitution.Set(ctx, "This chain has no constitution."); err != nil {
    56:				return err
    57:			}
    58:		}
    59:	
=>  60:		return nil

```

https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/migrations/v5/store.go#L56-L60

The rest ran, you can check by:

```
osmosisd q gov params --node https://rpc.osmosis.zone:443 --output json | jq
```